src/stages/mining_create_block.rs

```diff
@@ -0,0 +1,172 @@
+                number: BlockNumber(0),
+                gas_limit: 0,
+                gas_used: 0,
+                timestamp: 0,
+                extra_data: Bytes::new(),
+                mix_hash: H256::zero(),
+                nonce: H64::zero(),
+                base_fee_per_gas: None,
+            },
+            ommers: vec![],
+            transactions: vec![],
+        };
+        let mining_block_mutex = Arc::new(Mutex::new(mining_block));
+        self.mining_block = Arc::clone(&mining_block_mutex);
+        let parent_number = input.stage_progress.unwrap();
+        let parent_header = get_header(tx, parent_number)?;
+
+        // TODO, complete the remaining tx related part after txpool ready.
+        let mut proposal = create_block_header(&parent_header, Arc::clone(&self.mining_config))?;
+        debug!("Empty block created: {:?}", proposal);
+
+        let buffer = Buffer::new(tx, None);
+        if self.chain_spec.consensus.is_parlia() {
+            assert!(self
+                .mining_config
+                .lock()
+                .unwrap()
+                .consensus
+                .prepare(&buffer, &mut proposal)
+                .is_ok());
+
+            // If we are care about TheDAO hard-fork check whether to override the extra-data or not
+            if let Some(dao_block) = &self.mining_config.lock().unwrap().dao_fork_block {
+                // Check whether the block is among the fork extra-override range
+                let limit = BigInt::checked_add(&dao_block, &BigInt::from(DAOFORKEXTRARANG));
+                if proposal.number.0.cmp(&DAOFORKEXTRARANG.to_u64().unwrap()) >= Ordering::Equal
+                    && proposal.number.0.cmp(&limit.unwrap().to_u64().unwrap()) == Ordering::Less
+                {
+                    let dao_fork_block_extra =
+                        hex::decode("0x64616f2d686172642d666f726b").unwrap().into();
+                    // Depending whether we support or oppose the fork, override differently
+                    if self.mining_config.lock().unwrap().dao_fork_support {
+                        proposal.extra_data = dao_fork_block_extra;
+                    } else if bytes::Bytes::eq(&proposal.extra_data, &dao_fork_block_extra) {
+                        // If miner opposes, don't let it use the reserved extra-data
+                        proposal.extra_data = bytes::Bytes::default();
+                    }
+                };
+            }
+        }
+
+        self.mining_block.lock().unwrap().header = proposal;
+        debug!("Miner created block with data");
+
+        Ok(ExecOutput::Progress {
+            stage_progress: parent_number + 1,
+            done: true,
+            reached_tip: true,
+        })
+    }
+
+    async fn unwind<'tx>(
+        &mut self,
+        _tx: &'tx mut MdbxTransaction<'db, RW, E>,
+        _input: UnwindInput,
+    ) -> anyhow::Result<UnwindOutput>
+    where
+        'db: 'tx,
+    {
+        todo!()
```

Comment:
notice: todo!() will panic when call here, you can leave TODO comment? or add some warn logs.